src/dispatching/update_listeners/polling.rs

```diff
@@ -1,89 +1,192 @@
+        Self { drop_pending_updates: true, ..self }
+    }
+
+    /// Deletes webhook if it was set up.
+    pub async fn delete_webhook(self) -> Self {
+        delete_webhook_if_setup(&self.bot).await;
+
+        self
+    }
+
+    /// Returns a long polling update listener with configuration from the
+    /// builder.
+    ///
+    /// See also: [`polling_default`], [`Polling`].
+    pub fn build(self) -> Polling<R> {
+        let Self { bot, timeout, limit, allowed_updates, drop_pending_updates } = self;
+        let (token, flag) = AsyncStopToken::new_pair();
+        Polling { bot, timeout, limit, allowed_updates, drop_pending_updates, flag, token }
+    }
+}
+
 /// Returns a long polling update listener with `timeout` of 10 seconds.
 ///
-/// See also: [`polling`](polling).
+/// See also: [`Polling::builder`].
 ///
 /// ## Notes
 ///
 /// This function will automatically delete a webhook if it was set up.
-pub async fn polling_default<R>(requester: R) -> impl UpdateListener<R::Err>
+pub async fn polling_default<R>(bot: R) -> Polling<R>
 where
     R: Requester + Send + 'static,
     <R as Requester>::GetUpdates: Send,
 {
-    delete_webhook_if_setup(&requester).await;
-    polling(requester, Some(Duration::from_secs(10)), None, None)
+    Polling::builder(bot).timeout(Duration::from_secs(10)).delete_webhook().await.build()
 }
 
-#[cfg_attr(doc, aquamarine::aquamarine)]
 /// Returns a long polling update listener with some additional options.
-///
-/// - `bot`: Using this bot, the returned update listener will receive updates.
-/// - `timeout`: A timeout in seconds for polling.
-/// - `limit`: Limits the number of updates to be retrieved at once. Values
-///   between 1—100 are accepted.
-/// - `allowed_updates`: A list the types of updates you want to receive.
-///
-/// See [`GetUpdates`] for defaults.
-///
-/// See also: [`polling_default`](polling_default).
-///
-/// ## Notes
-///
-/// - `timeout` should not be bigger than http client timeout, see
-///   [`default_reqwest_settings`] for default http client settings.
-/// - [`repl`]s and [`Dispatcher`] use [`hint_allowed_updates`] to set
-///   `allowed_updates`, so you rarely need to pass `allowed_updates`
-///   explicitly.
-///
-/// [`default_reqwest_settings`]: teloxide::net::default_reqwest_settings
-/// [`repl`]: fn@crate::repl
-/// [`Dispatcher`]: crate::dispatching::Dispatcher
-/// [`hint_allowed_updates`]:
-/// crate::dispatching::update_listeners::UpdateListener::hint_allowed_updates
+#[deprecated(since = "0.10.0", note = "use `Polling::builder()` instead")]
+pub fn polling<R>(
+    bot: R,
+    timeout: Option<Duration>,
+    limit: Option<u8>,
+    allowed_updates: Option<Vec<AllowedUpdate>>,
+) -> Polling<R>
+where
+    R: Requester + Send + 'static,
+    <R as Requester>::GetUpdates: Send,
+{
+    let mut builder = Polling::builder(bot);
+    builder.timeout = timeout;
+    builder.limit = limit;
+    builder.allowed_updates = allowed_updates;
+    builder.build()
```

Comment:
We could make fields of the builder public, so users could do something similar, if needed.

Reply:
But why are you using fields here instead of the methods?

Reply:
Because methods accept `T` and I have `Option<T>`.

Reply:
Got it. Let's export these fields then.

Reply:
Done in cd63dbf1e7d76bd98bfa239e09c357c76d97cb4e